base/regex.jl

```diff
@@ -15,6 +15,8 @@ type Regex
     extra::Ptr{Void}
     ovec::Vector{Csize_t}
     match_data::Ptr{Void}
+    capture_name_to_idx::Dict{Symbol, Int}
+    idx_to_capture_name::Dict{Int, Symbol}
```

Comment:
Alright, I'm no longer caching the capture names in the regex table. The only time memory is allocated now is when ``show`` is called on a ``Match`` object.